source/Calamari/Commands/RunScriptCommand.cs

```diff
@@ -87,7 +89,7 @@ public override int Execute(string[] commandLineArguments)
                 new ExecuteScriptConvention(scriptEngine, commandLineRunner)
             };
 
-            var deployment = new RunningDeployment(packageFile, variables);
+            var deployment = runningDeploymentFactory(packageFile);
```

Comment:
Using a factory here isn't really justified, IMO, and overcomplicates this.  The constructor only has two parameters (three in other places), and the `variables` field is needed elsewhere in this class anyway.
It also means you have to create the factory in the tests manually.

Reply:
That's fair - I have another PR to make this use dependency injection properly which I'll do as a separate PR.

Reply:
Oh I just realised why I did it - I've done it this way so I can slip in Environment Variables in the test - if it's `new`ed up in RunScriptCommand I have no way of passing environment variables to it. This way I can setup the factory in my tests to pass in the environment variables when it creates the instance.